AdminClient/src/app/supplier/store/suppliers.effects.ts

```diff
@@ -0,0 +1,145 @@
+import { Action, Store } from '@ngrx/store';
+import { of } from 'rxjs';
+import { delay, map, switchMap, take, tap } from 'rxjs/operators';
+import * as fromApp from '../../reducers/index';
+import { PopupDialogComponent } from '../../shared/components/popup-dialog/popup-dialog.component';
+import { PopupDialogDataModel } from '../../shared/model/popup-dialog-data.model';
+import { SupplierModel } from '../../shared/model/supplier/supplier.model';
+import { SupplierFormComponent } from '../supplier-form/supplier-form.component';
+import * as SupplierActions from './suppliers.actions';
+import * as fromSuppliers from './suppliers.reducer';
+
+@Injectable()
+export class SuppliersEffects {
+  deleteCustomer$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.deleteSupplier),
+      switchMap((action: Action) => of(this.deleteSupplier()).pipe(
+          delay(2000),
+          tap(() => this.dialog.closeAll())
+      ))
+  ));
+
+  editSupplier$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.editSupplier),
+      switchMap((action: Action & { editedSupplier: SupplierModel }) => {
+        return of(this.editCustomer(action.editedSupplier)).pipe(delay(2000));
+      })
+  ));
+
+  showDeleteSupplierDialog$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.showDeleteSupplier),
+      map((action: Action & { focusedSupplier: SupplierModel }) => {
+        this.dialog.open<PopupDialogComponent, PopupDialogDataModel>(
+            PopupDialogComponent,
+            {
+              data: {
+                title: 'Delete supplier',
+                description: 'Are you sure you want to delete this supplier?',
+                dialogActions: [
+                  {
+                    text: 'Confirm',
+                    action: () => this.store$.dispatch(SupplierActions.deleteSupplier()),
+                    color: 'warn'
+                  },
+                  {
+                    text: 'Cancel',
+                    action: () => this.dialog.closeAll(),
+                    color: 'primary'
+                  }
+                ],
+                isDisabled: this.store$.select(fromSuppliers.selectIsLoading)
+              }
+            }
+        );
+      })
+  ), {dispatch: false});
+
+  showEditSupplierDialog$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.showEditSupplier),
+      map((action: Action & { focusedSupplier: SupplierModel }) => {
+        this.dialog.open<SupplierFormComponent, SupplierModel>(SupplierFormComponent,
+            {
+              width: '66vw',
+              height: '66vh',
+              panelClass: 'customerDialog'
+            }
+        );
+      })
+  ), {dispatch: false});
+
+  loadAllClients$ = createEffect(() => this.actions$.pipe(
+      ofType(SupplierActions.beginLoadingSuppliers),
+      switchMap((action: Action) => of(this.loadAllClients()).pipe(delay(2000)))
```

Comment:
```suggestion
      switchMap((action: Action) => of(this.loadAllSuppliers()).pipe(delay(2000)))
```

Check References too